buildbot/master/files/config/factories.py

```diff
@@ -109,6 +113,17 @@ def make_step(self, command):
             elif arg == './etc/ci/upload_nightly.sh':
                 step_kwargs['logEnviron'] = False
                 step_env += envs.upload_nightly
+                if self.is_windows:
+                    # s3cmd on Windows only works within msys
+                    step_env['MSYSTEM'] = 'MSYS'
```

Comment:
Yeah, it's already in deployment as-formatted, so I'm kinda tempted to get this landed (since I already feel bad about having an ad-hoc branch in production).